include/mqt-core/dd/GateMatrixDefinitions.hpp

```diff
@@ -10,31 +10,27 @@
 #pragma once
 
 #include "dd/DDDefinitions.hpp"
+#include "ir/operations/OpType.hpp"
 
 #include <cmath>
 #include <complex>
+#include <unordered_map>
+#include <vector>
 
 namespace dd {
 
 // Gate matrices
-constexpr GateMatrix I_MAT{1, 0, 0, 1};
-constexpr GateMatrix H_MAT{SQRT2_2, SQRT2_2, SQRT2_2, -SQRT2_2};
-constexpr GateMatrix X_MAT{0, 1, 1, 0};
-constexpr GateMatrix Y_MAT{0, {0, -1}, {0, 1}, 0};
-constexpr GateMatrix Z_MAT{1, 0, 0, -1};
-constexpr GateMatrix S_MAT{1, 0, 0, {0, 1}};
-constexpr GateMatrix SDG_MAT{1, 0, 0, {0, -1}};
-constexpr GateMatrix T_MAT{1, 0, 0, {SQRT2_2, SQRT2_2}};
-constexpr GateMatrix TDG_MAT{1, 0, 0, {SQRT2_2, -SQRT2_2}};
-constexpr GateMatrix SX_MAT{
-    std::complex{0.5, 0.5}, {0.5, -0.5}, {0.5, -0.5}, {0.5, 0.5}};
-constexpr GateMatrix SXDG_MAT{
-    std::complex{0.5, -0.5}, {0.5, 0.5}, {0.5, 0.5}, {0.5, -0.5}};
-constexpr GateMatrix V_MAT{SQRT2_2, {0, -SQRT2_2}, {0, -SQRT2_2}, SQRT2_2};
-constexpr GateMatrix VDG_MAT{SQRT2_2, {0, SQRT2_2}, {0, SQRT2_2}, SQRT2_2};
 constexpr GateMatrix MEAS_ZERO_MAT{1, 0, 0, 0};
 constexpr GateMatrix MEAS_ONE_MAT{0, 0, 0, 1};
 
+extern const std::unordered_map<qc::OpType,
+                                GateMatrix (*const)(const std::vector<fp>&)>
+    MATS_GENERATORS;
+
+extern const std::unordered_map<qc::OpType, TwoQubitGateMatrix (*const)(
+                                                const std::vector<fp>&)>
+    TWO_GATE_MATS_GENERATORS;
+
 inline GateMatrix uMat(const fp lambda, const fp phi, const fp theta) {
```

Comment:
I would argue that the definitions of these methods for gate matrices could be completely moved to the `.cpp` file.
They shouldn't actually be needed outside of the gate matrix generators. Might as well encapsulate them there.
Any uses of the existing methods can just use the factory methods.